packages/core/src/index.ts

```diff
@@ -167,23 +176,19 @@ export async function Auth(
     if (request.method === "POST" && internalRequest.action === "session")
       return Response.json(null, { status: 400 })
 
-    const type = isAuthError ? error.type : "Configuration"
-    const page = (isAuthError && error.kind) || "error"
-    // TODO: Filter out some error types from being sent to the client
+    const isClientSafeErrorType = isClientError(error)
+    const type = isClientSafeErrorType ? error.type : "Configuration"
+
     const params = new URLSearchParams({ error: type })
-    const path =
-      config.pages?.[page] ?? `${config.basePath}/${page.toLowerCase()}`
+    if (error instanceof CredentialsSignin) params.set("code", error.code)
 
-    const url = `${internalRequest.url.origin}${path}?${params}`
+    const pageKind = (isAuthError && error.kind) || "error"
+    const pagePath = config.pages?.[pageKind] ?? `/${pageKind.toLowerCase()}`
+    const url = `${internalRequest.url.origin}${config.basePath}${pagePath}?${params}`
 
     if (isRedirect) return Response.json({ url })
-
     return Response.redirect(url)
```

Comment:
Hi, I'm having trouble managing to get the `res.redirect` URL when using a server action in a magic link

My action looks like this in my custom signin page
``` javascript
import { signIn } from '$/auth'

      <form
        action={async (formData) => {
          'use server'
          await signIn('nodemailer', {
            callbackUrl,
            email: formData.get('email')?.toString()!,
          })
        }}
      >
```

Reply:
Debugging I see the issue is on `@auth/core` in this package in combination with `next-auth` package. This `Response.redirect(url)` is received here:
https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/lib/actions.ts#L76

```javascript
  const res = await Auth(req, { ...config, raw, skipCSRFCheck })

  // ... other code

  if (shouldRedirect) return redirect(res.redirect!)
```

This is my case. `res.redirect` is `undefined` 
and I see this when an error occur with Nodemailer (I misconfigure it to see this error)

<img width="1313" alt="image" src="https://github.com/nextauthjs/next-auth/assets/49499/a0f9e857-8385-478a-892a-9a933138b164">

Reply:
I added a `logger.debug` and I can see that the right Error URL is there in this file but when doing `Response.redirect(url)` is received a `undefenid` in `next-auth/src/lib/actions.ts`